.github/workflows/int-and-connected-test-run.yml

```diff
@@ -158,138 +90,98 @@ jobs:
         ports:
           - 5432:5432
         options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+    env:
+      # this must be defined for the bash redirection
+      GOOGLE_APPLICATION_CREDENTIALS: 'jade-dev-account.json'
+      # this must be defined for the bash redirection
+      GOOGLE_SA_CERT: 'jade-dev-account.pem'
+      # required for locking and deployment to integration namespace
+      K8_CLUSTER: 'integration-master'
     steps:
-      - name: "Checkout code"
-        uses: actions/checkout@v3
-      - name: "Cache build"
-        uses: actions/cache@v3
-        with:
-          path: ${{ env.CACHE_PATHS }}
-          key: ${{ runner.os }}-build-integration
-      - name: "Whitelist Runner IP"
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+          cache: 'gradle'
+      - name: Whitelist Runner IP
         uses: broadinstitute/datarepo-actions/actions/main@0.74.0
         with:
           actions_subcommand: 'gcp_whitelist'
           sa_b64_credentials: ${{ secrets.SA_B64_CREDENTIALS }}
-      - name: "Check for an available namespace to deploy API to and set state lock"
+      - name: Check for an available namespace to deploy API to and set state lock
         uses: broadinstitute/datarepo-actions/actions/main@0.74.0
         with:
           actions_subcommand: 'k8_checknamespace'
           k8_namespaces: 'integration-1,integration-2,integration-3,integration-6'
           sa_b64_credentials: ${{ secrets.SA_B64_CREDENTIALS }}
-      - name: "Build docker container via Gradle"
+      - name: Build docker container via Gradle
         uses: broadinstitute/datarepo-actions/actions/main@0.74.0
+        env:
+          # unset the GitHub Action default JAVA_HOME to build with JDK 17
+          JAVA_HOME:
         with:
           actions_subcommand: 'gradlebuild' # creates gcr build with git_hash tag
           sa_b64_credentials: ${{ secrets.SA_B64_CREDENTIALS }}
-      - name: "Deploy to cluster with Helm"
+      - name: Deploy to cluster with Helm
         uses: broadinstitute/datarepo-actions/actions/main@0.74.0
         with:
           actions_subcommand: 'helmdeploy'
           helm_create_secret_manager_secret_version: 0.0.8
-          helm_datarepo_api_chart_version: 0.0.739
-          helm_datarepo_ui_chart_version: 0.0.360
+          helm_datarepo_api_chart_version: 0.0.744
+          helm_datarepo_ui_chart_version: 0.0.362
```

Comment:
I'm unfamiliar with how this works. How are these versions updated normally?

Reply:
There's a separate action that bumps these and keeps them synced across environments. It's another wart that will be removed once we transition away from the current integration testing approach.

Reply:
(note that I didn't manually update these values, these came from restoring the integration test piece)